web-ui/src/pages/CheckinsPage.jsx

```diff
@@ -1,22 +1,30 @@
-import React, { useContext, useState } from "react";
+import React, { useContext, useEffect, useState } from "react";
 import CheckinsHistory from "../components/checkin/CheckinHistory";
 import CheckinDocs from "../components/checkin/CheckinDocs";
 import Personnel from "../components/personnel/Personnel";
 import Modal from "../components/modal/Modal";
-import Button from "@material-ui/core/Button";
 import GuidesPanel from "../components/guides/GuidesPanel";
+import CheckinProfile from "../components/checkin/CheckinProfile";
 import Note from "../components/notes/Note";
 import { AppContext } from "../context/AppContext";
+
+import Button from "@material-ui/core/Button";
+
 import "./CheckinsPage.css";
 
-const CheckinsPage = () => {
+const CheckinsPage = ({ history }) => {
   const [show, setShow] = useState(false);
   const { state } = useContext(AppContext);
-  const { checkins, userProfile } = state;
-  const [index, setIndex] = useState(0);
+  const { checkins, index, userProfile } = state;
   const checkin = checkins[index];
   const canSeePersonnel = userProfile && userProfile.role.includes("PDL");
 
+  useEffect(() => {
+    if (checkin && checkin.id) {
+      history.push(`/checkins/${checkin.id}`);
```

Comment:
Ok...I think this may be a little bit backward. I think we should be pulling the checkin id off of the URL when it's there and respecting that. The history should control what the current check-in is by changing the URL. Otherwise, if I hit refresh while looking at a given check-in, I will likely lose what I was looking at and be shown the default.